### PR TITLE
add round-trip serde for prc/pdb and tests

### PR DESCRIPTION
**Pull request by knickish**
Tiny bit of cleanup, and lots of changes to the serialization and deserialization code with tests to match. Can now load and then output a byte-for-byte matching pdb/prc for all the examples already in the repo.

### CODE REVIEW

**Comment by u1f408** [comment]
Hey, thanks for the PR! I'll look at this properly in the next couple days, but from a quick glance this looks great :)

**Comment by knickish** [comment]
Sounds good! Planning to work on this quite a bit over the next couple months, will try to upstream what I can

File: palmrs-database/src/info/category.rs [hunk 1]
```diff
@@ -85,17 +104,58 @@ impl AppInfoCategories {
...
+		}
+
+		let last_unique_id = rdr.read_u8()?;
+		let rsvd = rdr.read_u8()?;
+
+		dbg!(&categories);
+		for category in &categories {
+			let res = category.name_try_str();
+			let _ = dbg!(res);
+		}
```
Comment (u1f408) [inline palmrs-database/src/info/category.rs]: remove `dbg!()` calls

File: palmrs-database/tests/format_pdb_palmdoc.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ const EXAMPLE_PDB: &'static [u8] = include_bytes!("../../test-data/tWmanual.pdb"
 
 #[test]
 fn read_header() {
-	let header = DatabaseHeader::from_bytes(&EXAMPLE_PDB).unwrap();
+	let header = dbg!(DatabaseHeader::from_bytes(&EXAMPLE_PDB).unwrap());
```
Comment (u1f408) [inline palmrs-database/tests/format_pdb_palmdoc.rs]: and here (not _as_ important bc this is only in a test, but...)

File: palmrs-database/tests/format_pdb_tododb.rs [hunk 3]
```diff
@@ -20,6 +20,7 @@ fn read_header() {
 #[test]
 fn read_database_full() {
 	let database = PalmDatabase::<PdbWithCategoriesDatabase>::from_bytes(&EXAMPLE_PDB).unwrap();
+	// dbg!(&database);
```
Comment (u1f408) [inline palmrs-database/tests/format_pdb_tododb.rs]: and here

**Review by u1f408 — CHANGES_REQUESTED** [review]
Looks great! My only real nitpick is the `dbg!()` calls you've left scattered in there, other than that I'm happy to merge this :)

**Comment by knickish** [comment]
I think I got them all. Was one more you hadn't listed here that I'd missed also.

**Review by u1f408 — APPROVED** [review]

**Comment by u1f408** [comment]
Thanks heaps! 😸

**Comment by knickish** [comment]
Of course! This repo saved me so much time, thought I was going to have to do it from scratch.

**Pull request merged**